Src/Web/Web.Shared.Net/ApplicationInsightsHttpModule.cs

```diff
@@ -92,13 +93,52 @@ private void OnBeginRequest(object sender, EventArgs eventArgs)
 
                 this.TraceCallback("OnBegin", httpApplication);
 
+                this.HookOnSendRequestHeaders(httpApplication);
+
                 if (this.requestModule != null)
                 {
                     this.requestModule.OnBeginRequest(httpApplication.Context);
                 }
 
                 // Kept for backcompat. Should be removed in 2.3 SDK
-                WebEventsPublisher.Log.OnBegin();    
+                WebEventsPublisher.Log.OnBegin();
+            }
+        }
+
+        /// <summary>
+        /// When sending the response headers, allow request module to add the IKey's target hash.
+        /// </summary>
+        /// <param name="httpApplication">HttpApplication instance.</param>
+        private void HookOnSendRequestHeaders(HttpApplication httpApplication)
+        {
+            try
+            {
+                if (httpApplication != null && httpApplication.Response != null)
+                {
+                    // We use reflection here because 'AddOnSendingHeaders' is only available post .net framework 4.5.2. Hence we call it if we can find it.
+                    // Not using reflection would result in MissingMethodException when 4.5 or 4.5.1 is present. 
+                    MethodInfo addOnSendingHeadersMethod = httpApplication.Response.GetType().GetMethod("AddOnSendingHeaders");
+
+                    if (addOnSendingHeadersMethod != null)
+                    {
+                        var parameters = new object[]
+                        {
+                            new Action<HttpContext>((httpContext) =>
+                            {
+                                if (this.requestModule != null)
```

Comment:
I still do not see try{}catch here. Runaway exception in this callback will fail the customer's code